### ISSUE TITLE
Document that *Source attributes permit Task<IEnumerable> return types

### ISSUE THREAD

**Issue by stevenaw** [nunit/docs#756]
Method-based *Source attributes gained support for async return types in upcoming v4 and v3.14 framework versions via https://github.com/nunit/nunit/issues/2729
It would be good to document this change. List of updated attributes can be found in https://github.com/nunit/nunit/pull/4389

**Comment by stevenaw** [comment]
@SeanKilleen Curious, how would something like this be approached for unreleased changes? Is there a "prerelease" section of the docs, or should a PR be put up and just not merged until the framework changes are released?

**Comment by OsirisTerje** [comment]
@stevenaw FWIW: I have been in the same situation for the adapter, and I chose to add a note to indicate from which version the feature was available.   E.g. see  https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#where

For future features I know which version it goes into, so I add that upcoming version number.

**Comment by SeanKilleen** [comment]
@stevenaw  yep, @OsirisTerje is correct here -- I'll see if anything changed but docfx is a little clunky here; it doesn't easily allow multiple versions of docs AFAIK.

**Comment by stevenaw** [comment]
Thanks @OsirisTerje @SeanKilleen !
I'll give that a try.

**Comment by stevenaw** [comment]
Spare time has been a bit hard to come by lately. I'm going to remove my assignment from this for now in case anyone else wishes to pick this up
